src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -480,13 +476,58 @@ private static ExpressionSyntax GetObjectCreationExpressionSyntax(Type type)
             else
             {
                 // Create an unformatted object.
-                Expression<Func<object>> getUninitializedObject =
+#if DNXCORE50
+                var bindingFlags = SyntaxFactoryExtensions.GetBindingFlagsParenthesizedExpressionSyntax(
+                    SyntaxKind.BitwiseOrExpression,
+                    BindingFlags.Instance,
+                    BindingFlags.NonPublic,
+                    BindingFlags.Public);
+                var nullLiteralExpressionArgument = SF.Argument(SF.LiteralExpression(SyntaxKind.NullLiteralExpression));
+                var typeArg = SF.Argument(SF.TypeOfExpression(typeInfo.GetTypeSyntax()));
+                var bindingFlagsArg = SF.Argument(bindingFlags);
+                var binderArg = nullLiteralExpressionArgument;
+                ArgumentSyntax ctorArgumentsArg;
+                var cons = typeInfo.GetConstructors(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance)
+                    .OrderBy(p => p.GetParameters().Count()).FirstOrDefault();
+                var consParameters = cons != null ? cons.GetParameters().Select(p => p.ParameterType).ToArray() : null;
+                if (consParameters != null && consParameters.Length > 0)
+                {
+                    var separatedSyntaxList = new SeparatedSyntaxList<ExpressionSyntax>();
+                    separatedSyntaxList = consParameters
+                        .Aggregate(separatedSyntaxList, (current, t) => current.Add(SF.DefaultExpression(t.GetTypeInfo().GetTypeSyntax())));
+                    ctorArgumentsArg = SF.Argument(separatedSyntaxList.GetArrayCreationWithInitializerSyntax(
+                        SF.PredefinedType(SF.Token(SyntaxKind.ObjectKeyword))));
+                }
+                else
+                {
+                    ctorArgumentsArg = nullLiteralExpressionArgument;
+                }
+
+                var cultureInfoArg = SF.Argument(
+                    SF.IdentifierName("System").Member("Globalization").Member("CultureInfo").Member("InvariantCulture"));
+                var createInstanceArguments = new []
+                {
+                    typeArg,
+                    bindingFlagsArg,
+                    binderArg,
+                    ctorArgumentsArg,
+                    cultureInfoArg
+                };
+
+                Expression<Func<object>> getUninitializedObject = () => Activator.CreateInstance(default(Type));
```

Comment:
While this will work for getting the method, because of how the `.Invoke()` extension method works, it would be good to use the expression for the actual constructor you're calling.